### PR TITLE
Save EntityStarted when StartEntity requested via remembered entities

### LINKED ISSUE #25809: Sharding forgets remembered entities with change in sharding funcion
When you have sharding with remembered entities and you change sharding function when cluster is not running. What happens with entities, that should now go to different shard? It looks like those will start, but will be forgotten.

The original shard will remove the information about an entity here:
https://github.com/akka/akka/blob/abb3429bc876f5659bd65adf58642c66f42ffe06/akka-cluster-sharding/src/main/scala/akka/cluster/sharding/Shard.scala#L186

And the new shard will just start it:
https://github.com/akka/akka/blob/abb3429bc876f5659bd65adf58642c66f42ffe06/akka-cluster-sharding/src/main/scala/akka/cluster/sharding/Shard.scala#L179

but will not remember it via `EntityStarted`

Or am I missing something?

### PR DESCRIPTION
**Pull request by chbatey**
Fixes #25809

### CODE REVIEW

File: akka-cluster-sharding/src/main/scala/akka/cluster/sharding/Shard.scala [hunk 1]
```diff
@@ -170,14 +170,14 @@ private[akka] class Shard(
...
     case RestartEntities(ids) ⇒ restartEntities(ids)
   }
 
   def receiveStartEntity(start: ShardRegion.StartEntity): Unit = {
-    log.debug("Got a request from [{}] to start entity [{}] in shard [{}]", sender(), start.entityId, shardId)
-    getEntity(start.entityId)
-    sender() ! ShardRegion.StartEntityAck(start.entityId, shardId)
+    val requester = sender()
+    log.debug("Got a request from [{}] to start entity [{}] in shard [{}]", requester, start.entityId, shardId)
+    getOrCreateEntity(start.entityId, _ ⇒ processChange(EntityStarted(start.entityId))(_ ⇒ requester ! ShardRegion.StartEntityAck(start.entityId, shardId)))
```
Comment (patriknw) [inline akka-cluster-sharding/src/main/scala/akka/cluster/sharding/Shard.scala]: ok, `processChange` is the fix

**Review by patriknw — APPROVED** [review]
LGTM, optional nitpicks

**Review by johanandren — APPROVED** [review]
Nothing to add on top of Patriks nits, LGTM

**Comment by patriknw** [comment]
Binary compatibility check failed

**Comment by patriknw** [comment]
Added mima filter in https://github.com/akka/akka/pull/26061

**Pull request closed without merging**